DataFormats/Math/interface/deltaPhi.h

```diff
@@ -59,4 +61,76 @@ struct DeltaPhi {
   }
 };
 
+
+namespace angle_units {
+  
+  constexpr long double piRadians(M_PIl);  // M_PIl is long double version of pi
+  constexpr long double degPerRad = 180. / piRadians; // Degrees per radian
+  
+  namespace operators {
+
+    // Angle
+    constexpr long double operator "" _pi( long double x ) 
+    { return x * piRadians; }
+    constexpr long double operator "" _pi( unsigned long long int x ) 
+    { return x * piRadians; }
+    constexpr long double operator"" _deg( long double deg )
+    {
+      return deg / degPerRad;
+    }
+    constexpr long double operator"" _deg( unsigned long long int deg )
+    {
+      return deg / degPerRad;
+    }
+    constexpr long double operator"" _rad( long double rad )
+    {
+      return rad * 1.;
+    }
+
+    template <class NumType>
+    inline constexpr NumType convertRadToDeg(NumType radians) // Radians -> degrees
+    {
+      return (radians * degPerRad);
+    }
+
+    template <class NumType>
+    inline constexpr long double convertDegToRad(NumType degrees) // Degrees -> radians
+    {
+      return (degrees / degPerRad);
+    }
+  }
+}
+
+
+namespace angle0to2pi {
+
+  using angle_units::operators::operator""_pi;
+
+  // make0To2pi constrains an angle to be >= 0 and < 2pi.
+  // This function is a faster version of reco::reduceRange.
+  // In timing tests, it is almost always faster than reco::reduceRange.
+  // It also protects against floating-point value drift over repeated calculations.
+  // This implementation uses multiplication instead of division and avoids
+  // calling fmod to improve performance.
+
+  template <class valType>
+  inline constexpr valType make0To2pi(valType angle) {
```

Comment:
commenting here instead of in #27637 

about angle0to2pi
IMHO, it would be more consistent to move it up in this file to the reco namespace and rename to ```reduceRange0To2pi```
Also, if it's implementation is really faster, replace the current ```reduceRange```  of [-pi,pi] with the faster variant.

Reply:
@slava77 and @cvuosalo - can we think of other then reco namespace? It would be confusing if the DD code uses reco::

Reply:
Concerning the possible replacement of `reduceRange`: Since `make0To2pi` checks for floating-point value drift over repeated calculations and `reduceRange` does not, there is a good possibility that comparison tests for replacing `reduceRange` would show small differences. `make0To2pi` should provide more correct results, but it would be required to show that the comparison test differences were correct, appropriate, and desirable, which might be a challenge. I did not replace `reduceRange` in this PR for this very reason.

Perhaps replacement could be done on a step-by-step basis by developers familiar with the applications getting changed so that the comparison differences would be easier to understand.

Reply:
I'd suggest putting everything in the file in the `delta_phi` namespace to match the header file name, but I don't have a strong opinion about namespace naming.

Reply:
About the small differences, it should be acceptable, considering the expected speedup and improved numerical precision (unless perhaps a PR comes after the deadline with an "urgent" tag and some unrelated changes inside).